docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+# Simperby: Protocol Overview
+
+Simperby is a blockchain engine that runs an organiziation which is
+
+1. **Decentralized**
+2. **Standalone, Soverign and Self-hosted**
+3. **With a permissioned set of memebers**
+
+in that it provides
+
+1. **Consensus**: a safe and live mechanism to finalize the state of the organization, which is tolerant to byzantine faults in an asynchronous network.
+2. **Governance**: a democratic mechanism to make decisions on the organization, which can be finalized by the consensus.
+3. **Communication Channel**: a mechanism to communicate with each other in a decentralized, verifable, fault-tolerant way.
+
+## Basics and Terminologies
+
+The reader of this document should be familiar with the basic concepts of blockchain.
+
+- Simperby is an engine that can build an instnace of a blockchain.
+  Any blockchain built with Simperby is sometimes called a *Simperby chain*.
+- The unit of state transition is called a *transaction*.
+  `StoreData`, `DelegateGovernanceVotingPower`, `AddValidator` are the examples.
+- A sequence of a transactions is called an *agenda*.
+  **Any govenrnance voting is performed on an agenda and the target block height**.
+- A *block* consists of the header and **a single agenda, which is mandatory**.
+- The participants of the govenance are called *members*.
+- The participants of the consensus are called *validators*, which is a subset of the *members*.
+- A *consensus leader* or a *block proposer* is a validator that proposes a block for the round.
+- A *round* is a period of time in which a consensus leader proposes a block, and consensus votes(pre-vote, pre-commit) are performed. Refer to Tendermint for more details.
+- A validator is *honest* or *not byzantine* if they follow the consenus protocol.
+- A block proposer is *faithful* if they fulfill its responsibility, and *good* if they don't overuse their power. They are *lazy* and *bad* otherwise.
```

Comment:
```suggestion
- A block proposer is *faithful* if they fulfill their responsibility and *good* if they don't overuse their power. They are *lazy* and *bad* otherwise.
```